### PR TITLE
JDK-8281671: Class.getCanonicalName spec should explicitly cover array classes

### PR DESCRIPTION
**Pull request by jddarcy**
Add explicit discussion of array types to Class.getCanonicalName, no change in behavior.

I didn't see any existing regression tests for the various "getFooName" methods of Class so I created a new test file for a handful of test cases.

Please also review the corresponding CSR: https://bugs.openjdk.java.net/browse/JDK-8281922

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed
- [x] Change requires a CSR request to be approved

### Issues
 * [JDK-8281671](https://bugs.openjdk.java.net/browse/JDK-8281671): Class.getCanonicalName spec should explicitly cover array classes
 * [JDK-8281922](https://bugs.openjdk.java.net/browse/JDK-8281922): Class.getCanonicalName spec should explicitly cover array classes (**CSR**)

### Reviewers
 * [Mandy Chung](https://openjdk.java.net/census#mchung) (@mlchung - **Reviewer**) ⚠️ Review applies to 2b0729ad163c13fc6cbee90a0f896b40b7567605

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7483/head:pull/7483` \
`$ git checkout pull/7483`

Update a local copy of the PR: \
`$ git checkout pull/7483` \
`$ git pull https://git.openjdk.java.net/jdk pull/7483/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7483`

View PR using the GUI difftool: \
`$ git pr show -t 7483`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7483.diff">https://git.openjdk.java.net/jdk/pull/7483.diff</a>

</details>

### CODE REVIEW

File: test/jdk/java/lang/Class/NameTest.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8281671
+ * @summary Checks on various "getFooName" methods of java.lang.Class
```
Comment (mlchung) [inline test/jdk/java/lang/Class/NameTest.java:27]: This tests only `getCanonicalName` method.

File: test/jdk/java/lang/Class/NameTest.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        expectedCanonicalName.put(o.getClass(),                 null);
+
+        // If a component type doesn't have a canonical name, neither
+        // does an array of that type.
+        expectedCanonicalName.put(LocalClass.class.arrayType(), null);
+        expectedCanonicalName.put(o.getClass().arrayType(),     null);
+
+        expectedCanonicalName.put(int.class,              "int");
+        expectedCanonicalName.put(Object.class,           "java.lang.Object");
+        expectedCanonicalName.put(objectArray.getClass(), "java.lang.Object[]");
```
Comment (mlchung) [inline test/jdk/java/lang/Class/NameTest.java:56]: Just an observation: TestNG is a good option to clearly show the data provider and the test body.  Up to you.

**Review by mlchung — APPROVED** [review]

File: test/jdk/java/lang/Class/NameTest.java (hunk 1 above)
Comment (jddarcy) [reply test/jdk/java/lang/Class/NameTest.java:27]: Added analogous test cases for getSimpleName name.

File: test/jdk/java/lang/Class/NameTest.java (hunk 2 above)
Comment (jddarcy) [reply test/jdk/java/lang/Class/NameTest.java:57]: Noted; I'll stick with the hand-written code for now.

**Pull request closed without merging**